runner/custom_template_factory.py

```diff
@@ -62,6 +62,60 @@ def set_custom_image(in_memory_json_object: str, VM_image_URL: str, VM_image_typ
     # TODO For handling windows and centos the nodeAgentSKUId needs to be changed in the json send object
     # The centos image hasn't been tested so I haven't written logic for handling the VM_image_type
 
+def set_pool_resource_file(in_memory_json_object: str, resource_branch_name: str):
+    """
+    Replaces the github branch name to point to a new branch. 
+    "https://raw.../master/..../install-azure-nc-drivers.cmd",
+    "https://raw.../'resource_branch_name'/..../install-azure-nc-drivers.cmd",
+
+    This is for testing out new ps1 or bat files
+
+    :param in_memory_json_object: The json object that needs to be updated with a repo file path
+    :type in_memory_json_object: str
+    :param resource_branch_name: The value that needs to be set
+    :type resource_branch_name: str
+
+    """
+    if not resource_branch_name:
+        return
+
+    if in_memory_json_object.get("pool") is not None: 
+        if in_memory_json_object.get("pool").get("startTask") is not None: 
+            if in_memory_json_object["pool"]["startTask"]["resourceFiles"] is not None:
+                for index in range(len(in_memory_json_object.get("pool").get("startTask").get("resourceFiles"))):
+                    resource_file = in_memory_json_object.get("pool").get("startTask").get("resourceFiles")[index]
+                    for key in resource_file:
+                        resource_file[key] = resource_file[key].replace("/master", "/"+resource_branch_name.strip())
```

Comment:
change this to replace: "batch-extension-templates/master/"

as the way you have it, if the resource files is called /master-startup-script.ps1

you will replace that as well.